### DISCUSSION TITLE
How does overheat vary between guns?

### DISCUSSION THREAD

**Discussion by DJ-Reticuli** [acemod/ACE3#10605] [Q&A] [answered]
Ignoring the common ACE overheating settings all guns would be subjected to, is overheating of the gun varying by gun based on just the gun itself or the magazine or the magazine + barrel size?  I put an MX 50 cal mod into the Virtus MCX Spear that's glitchy and too big, but I figured it would be more realistic looking with 50 cal rounds.  Definitely looks the part but also doesn't seem to overheat like the MX 50.  A lot of people hated the overheating, but I kinda loved it, because it was a negative attribute on something that otherwise was all positives.

**Comment by DJ-Reticuli** [comment]
Some guns overheat very easily, while some won't overheat at all. Very strange. I can't tell in the gun mods what causes this or how to tweak it.

**Comment by Drofseh** [comment, marked as answer]
The amount of heat generated is based on the ammunition, the faster a bullet is and the heavier it is the more heat is generated per shot.
How quickly the gun heats up is based the mass of the gun, the heavier a gun is the more heat is required to raise its temperature.

A fast heavy bullet generating a lot of heat in a lightweight gun that doesn't have much ability to soak said heat will quickly find itself at a high temperature.

**Comment by DJ-Reticuli** [comment]
Yet, a small Raptor shooting subsonic 8.6blk is heating up super-fast, and a .338 NM Spear is not heating up much at all, and I can't find any rhyme or reason in either the gun or ammo pbos.  I'd like to decrease the overheat in the former, and increase it in the latter.

**Reply by rautamiekka** [reply]
I think you mean the MMG variant (SIG Sauer MG 338) which is 338NM.

It seems the superior thermal transfer of the latter is due to the latter being forged steel instead of cast or milled. It seems forged steel has a very different structure. And since it's simultaneously stronger, you can use less of it, which is how the latter got lighter.

Sources:
- Comments at https://www.reddit.com/r/ForgottenWeapons/comments/17p09xt/mg338_is_very_huge/
- Some answers at https://www.quora.com/Why-do-forged-metallic-parts-have-a-higher-rate-of-thermal-expansion-and-contraction-than-the-casted-machined-one
  - https://www.quora.com/Why-are-forged-metals-more-ductile/answer/S-Sreeram-3 (the answer is paywalled, but you can see the important parts of it without)
  - Automated AI answer, copypasted here: https://gist.github.com/rautamiekka/5365a0cf183b0574c4fa47e3b31613fa

**Reply by DJ-Reticuli** [reply]
It's a lightweight Raptor firing 8.6blk subs and a heavier Spear firing 338 NM supers that I'm citing here, but there are lots of other examples.  Conversely, the rather heavy Shak-12 suppressed firing even heavier subs at its highest full auto fire rate doesn't seem to want to heat up unless I crank the Overheat values to crazy levels, and then while that gun suddenly starts overheating eventually, most other guns will then heat up too quickly.  Is there an easy way to change how overheating occurs for particular ammo and / or particular guns, because it's definitely not changing predictably based on bullet grain, muzzle velocity, and gun mass that I can tell, and I don't see any mention of "thermal", "heat", "over", "cookoff" in any of the pbos.

**Reply by Drofseh** [reply]
If you want to look at the code you can see how the heat is generated from the projectile here:
<https://github.com/acemod/ACE3/blob/master/addons/overheating/functions/fnc_overheat.sqf>

And how the heat energy gets applied to the weapon here:
<https://github.com/acemod/ACE3/blob/master/addons/overheating/functions/fnc_updateTemperature.sqf>

If the weapons don't have ACE specfic values for the mass of the bullet or the barrel then they are estimated based on the `hit` and `caliber` of the ammo, and the `mass` of the weapon.

**Reply by DJ-Reticuli** [reply]
I will look at them. Thanks.  

All the weapons (such as barrels) and ammo mentioned above are ACE compatible with relevant values. I don't mess with non-ACE stuff that's blank under ACE arsenal.

I am curious, though, how does Arma 3 with ACE calculate or "estimate" damage & armor penetration if they're not blank or blank, respectively?  I personally think the hit and caliber values are ridiculously crude and it should be deriving both soft injury and armor penetration from the ACE values themselves when it can, like grain, BC, RPM from twist, velocity, etc, not hit and caliber.  I've mentioned this elsewhere.  Obviously if there are no ACE values, damage and penetration should be relegating to the canned hit and caliber values then, like it appears overheat does, besides the minor gripe that "caliber" is an odd term for penetration in vanilla, but I digress.  Everyone previously has told me, though, that ACE doesn't bother with complicated damage and penetration calculations EVER, but now it turns out ACE uses complicated calculations for overheat?!

**Reply by Drofseh** [reply]
I am not super familiar with how ACE Medical works, but I believe ACE takes whatever damage the engine says was done and bases it's wounds off that.

**Comment by DJ-Reticuli** [comment]
Bizarre that the overheat coding with ACE would be more complicated than the damage and injury system.
